src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -410,55 +410,61 @@ private static ModuleInfo[] GetModuleInfos(int processId, bool firstModuleOnly)
                     if (moduleCount <= moduleHandles.Length) break;
                     moduleHandles = new IntPtr[moduleHandles.Length * 2];
                 }
-                List<ModuleInfo> moduleInfos = new List<ModuleInfo>();
 
-                int ret;
+                List<ModuleInfo> moduleInfos = new List<ModuleInfo>(moduleCount);
+                StringBuilder baseName = new StringBuilder(1024);
+                StringBuilder fileName = new StringBuilder(1024);
+
                 for (int i = 0; i < moduleCount; i++)
                 {
-                    try
+                    while (true)
```

Comment:
Oh, I see.  It's not that we want to return the first module we're able to construct, it's that we want to return the first module period, even if we couldn't actually load it (in which case we'd be returning an empty set).